ignite/metrics/mAP.py

```diff
@@ -0,0 +1,402 @@
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+
+from ignite.metrics.metric import Metric, reinit__is_reduced
+
+
+def iou(gt: torch.Tensor, dt: torch.Tensor, crowd: List) -> torch.Tensor:
+    m = len(dt)
+    n = len(gt)
+
+    ious = torch.zeros(m, n)
+
+    for g in range(n):
+        G = gt[g].tolist()
+        ga = G[2] * G[3]
+        iscrowd = crowd[g]
+        for d in range(m):
+            D = dt[d].tolist()
+            da = D[2] * D[3]
+            ious[m - d - 1, g] = 0
+            w = min(D[2] + D[0], G[2] + G[0]) - max(D[0], G[0])
+            h = min(D[3] + D[1], G[3] + G[1]) - max(D[1], G[1])
+            if w <= 0 or h <= 0:
+                continue
+            i = w * h
+            if iscrowd:
+                u = da
+            else:
+                u = da + ga - i
+            ious[m - d - 1, g] = i / u
+
+    return ious
+
+
+i_id = 0
```

Comment:
Is it possible to wrap in an enum inside the class ?

Reply:
I don't understand what are suggesting, can you please elaborate a bit?